plugins/mapper-hll/src/test/resources/rest-api-spec/test/mapper_hll/10_basic.yml

```diff
@@ -0,0 +1,81 @@
+# Integration tests for Mapper HLL components
+#
+
+---
+"Mapper HLL":
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            type1: { "properties": { "foo": { "type": "hll" } } }
+
+  - do:
+      index:
+        index: test
+        type: type1
+        id: 0
+        body: { "foo": "bar" }
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      search:
+        body: { "aggs": { "foo_count": { "cardinality": { "field": "foo" } } } }
+
+  - match: { aggregations.foo_count.value: 1 }
+
+  - do:
+      index:
+        index: test
+        type: type1
+        id: 1
+        body: { "foo": "bar" }
+
+  - do:
+      index:
+        index: test
+        type: type1
+        id: 2
+        body: { "foo": "baz" }
+
+  - do:
+      index:
+        index: test
+        type: type1
+        id: 3
+        body: { "foo": "bam" }
+
+  - do:
+      index:
+        index: test
+        type: type1
+        id: 4
+        body: { "foo": "bar" }
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      search:
+        body: { "aggs": { "foo_count": { "cardinality": { "field": "foo" } } } }
+
+  - match: { aggregations.foo_count.value: 3 }
+
+  - do:
+      index:
+        index: test
+        type: type1
+        id: 5
+        body: { "foo": {"items": ["1", "2", "3", "4"], "precision": 18} }
```

Comment:
This introduces a fielding test case until I can get `HLLFieldMapper` to fully support objects/arrays, but I'm already partway there as of d557530041feabd742d55fe9d24f8798cc4e2eb0